Assets/UIElements/_Elements/XRUI_Feedback/Scripts/XRUI_NearDetection.cs

```diff
@@ -30,12 +34,36 @@ void OnTriggerEnter(Collider other)
         void OnTriggerExit(Collider other)
         {
             XRController controller = other.GetComponent<XRController>();
+
+            if (controller == null)
+                controller = other.GetComponentInParent<XRController>();
+
             if (controller != null && IsNear)
             {
                 isNear = false;
                 feedbackBase.onNearExit?.Invoke(controller);
             }
         }
+
+        public void OnEnterAction(XRBaseInteractor xRBase)
+        {
+
+            XRController controller = xRBase.GetComponent<XRController>();
+            if (controller != null)
+            {
+                isNear = true;
+                feedbackBase.onNearEnter?.Invoke(controller);
+            }
+        }
+        public void OnExitAction(XRBaseInteractor xRBase)
+        {
+            XRController controller = xRBase.GetComponent<XRController>();
+            if (controller != null && IsNear)
+            {
+                isNear = false;
+                feedbackBase.onNearExit?.Invoke(xRBase.GetComponent<XRController>());
+            }
+        }
```

Comment:
when this functions are used?
The xrBase always will be contain a XRController?

Reply:
Changed